lib/layers/presentation/singup_user/singup_user_page.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:webview_flutter/webview_flutter.dart';
+
+import '../../../common/constants/constants.dart';
+import '../../../common/theme/text_styles.dart';
+import '../../domain/entities/user.dart';
+
+const String status = "status";
+class SingUpPage extends StatefulWidget {
+  const SingUpPage({Key? key}) : super(key: key);
+  static const id = "OpenId login";
+
+  @override
+  State<SingUpPage> createState() => _SingUpPageState();
+}
+
+class _SingUpPageState extends State<SingUpPage> {
+  late final WebViewController _controller;
+
+  @override
+  void initState() {
+
+    if (defaultTargetPlatform == TargetPlatform.android) WebView.platform = AndroidWebView();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    String realm = ModalRoute.of(context)!.settings.arguments as String;
+//todo find out why keyboard is lagging
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(SingUpPage.id, style: appBarTitle(context)),
+        actions: [
+          IconButton(
+            onPressed: () {
+              _controller.reload();
+            },
+            icon: const Icon(Icons.refresh),
+          )
+        ],
+      ),
+      body: WebView(
+        javascriptMode: JavascriptMode.unrestricted,
+        onWebViewCreated: (webViewController) {
+          _controller = webViewController;
+          _controller.clearCache();
+
+          final cookieManager = CookieManager();
+          cookieManager.clearCookies();
+        },
+        // userAgent: 'random',
+        initialUrl: (realm == EU) ? EU_LOGIN_URL : CIS_LOGIN_URL,
+        onPageStarted: (url) {
+          if (url.contains(REDIRECT_URL)) {
+            Map<String, String> response = Uri.splitQueryString(url);
+            String? status = response["status"];
+            if (status == "ok") {
+              User result = User(
```

Comment:
Why here we have not something like User.fromJson? If this is not json -- please at least extract to method